QUARKUS-658.md

```diff
@@ -0,0 +1,49 @@
+# QUARKUS-658 - Reactive REST client - support HTTP/2
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-658
+
+PR: https://github.com/quarkusio/quarkus/pull/31192
+
+Upstream issue: https://github.com/quarkusio/quarkus/issues/13969
+
+Quarkus should be able to use http/2 on client side to be able to comunicate with other microservices.
+
+## Scope of the testing
+Scope of the test will cover basic usage of rest client inside quarkus. The complex scenarios like sending request to diferent API/microservice won't be covered.
+
+Scope in point will be:
+- Run Quarkus application and send requests using `http/2` to verify that server response with `http/2`. 
+- Run Quarkus application with enabled Application-Layer Protocol Negotiation (alpn) to verify that correct http version is used.
+- Check the usecases using HTTP and HTTPS.
+
+These points will be covered for single REST client and all REST client to verify all of these properties:
+
+```
+// All REST clients
+quarkus.rest-client.http2=true
+quarkus.rest-client.alpn=true
+
+// single REST Client:
+quarkus.rest-client.extensions-api.http2=true
+quarkus.rest-client.extensions-api.alpn=true
+```
+
+New tests will be run in jvm and native mode.
+
+## Existing test coverage
+- We already have test for http/2 client side to test client sync and async. This test are both in reactive and non-reactive modules and are disabled.
+	- Non-reactive tests will be removed as only reactive client should be used.
+	- These test need to be updated as they failing.
+
+### Impact on test suites and testing automation
+- New module inside http module with name `http2`
```

Comment:
So the existing disabled tests will be moved to dedicated module?

Why not keeping the tests in https://github.com/quarkus-qe/quarkus-test-suite/tree/main/http/rest-client-reactive module? My preference would be not to introduce new module as the is already suitable "home" available.

Reply:
The existing test were in http-advanced. I'll move them to rest-client-reactive. I found out that we have this module on Tuesday morning. So now I update the test plan and all of the http2 tests coverage should be in rest-client-reactive module. Thanks for letting me know that we have it.